### PR TITLE
fix #11026 model config return

### LINKED ISSUE #11026: Chat conversations (in logs) cannot return the model config correctly
### Self Checks

- [X] This is only for bug report, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Dify version

0.11.2

### Cloud or Self Hosted

Cloud, Self Hosted (Docker), Self Hosted (Source)

### Steps to reproduce

image

### ✔️ Expected Behavior

It should return the model config correctly.

### ❌ Actual Behavior

It returns null

### PR DESCRIPTION
**Pull request by kenwoodjw**
# Summary

fix #11026 model config return

> [!Tip]
> Close issue syntax: `Fixes #<issue number>` or `Resolves #<issue number>`, see [documentation](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) for more details.

# Screenshots

<table>
  <tr>
  <td>Before: </td>
  <td>After: </td>
  </tr>
  <tr>
  <td>...</td>
  <td>...</td>
  </tr>
</table>

# Checklist

> [!IMPORTANT]  
> Please review the checklist below before submitting your pull request.

- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] I understand that this PR may be closed in case there was no previous discussion or issues. (This doesn't apply to typos!)
- [x] I've added a test for each change that was introduced, and I tried as much as possible to make a single atomic change.
- [x] I've updated the documentation accordingly.
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

### CODE REVIEW

File: api/models/model.py [hunk 1]
```diff
@@ -596,12 +596,14 @@ def model_config(self):
...
                 else:
                     model_config["configs"] = override_model_configs
+
             else:
                 app_model_config = (
                     db.session.query(AppModelConfig).filter(AppModelConfig.id == self.app_model_config_id).first()
                 )
                 if app_model_config:
                     model_config = app_model_config.to_dict()
+                    model_config["model_dict"] = app_model_config.model_dict
```
Comment (laipz8200) [inline api/models/model.py]: The property is already present in the 'model' object when we call the 'to_dict()' method. It appears that this issue cannot be resolved.

Comment (kenwoodjw) [reply api/models/model.py:606]: Got it, may be give me some tip?

Comment (kenwoodjw) [reply api/models/model.py]: I understand the issue now, as `model_config` corresponds to `model_config_fields`, not `simple_model_config_fields`. Perhaps this approach would work: 
```
simple_model_config_fields = {
    "model": fields.Raw(attribute="model"),
    "pre_prompt": fields.String,
}
```
What do you think？

**Comment by laipz8200** [comment]
image

In fact, we are currently losing the parameters in the completion_params. As you can see, this field only contains a 'stop' now, but the model parameters such as 'temperature' and 'top_p' should be contained in here.

**Comment by kenwoodjw** [comment]
@laipz8200  So the mode_config should return like that
```json
"model_config": {
                "model": {
                    "provider": "deepseek",
                    "name": "deepseek-chat",
                    "mode": "chat",
                    "completion_params": {
                        "stop": [],
                        "top_p": [],
                        "temperature": []
                    }
                },
                "pre_prompt": ""
            }
```
or simple ?

**Comment by laipz8200** [comment]
> @laipz8200 So the mode_config should return like that
> 
> ```json
> "model_config": {
>                 "model": {
>                     "provider": "deepseek",
>                     "name": "deepseek-chat",
>                     "mode": "chat",
>                     "completion_params": {
>                         "stop": [],
>                         "top_p": [],
>                         "temperature": []
>                     }
>                 },
>                 "pre_prompt": ""
>             }
> ```
> 
> or simple ?

Yes, here is the correct structure(from the front-end):

image

**Comment by kenwoodjw** [comment]
After my test, I opened the parameter options.
image

Returning the results we desire.
image

image

**Comment by laipz8200** [comment]
We found that is better to fix it at the frond-end, would you mind checking #11298?

**Comment by kenwoodjw** [comment]
> We found that is better to fix it at the frond-end, would you mind checking #11298?

Ok, I will close the PR

**Pull request closed without merging**